### PR TITLE
Added FaselHD Provider

### PR DESCRIPTION
**Pull request by ImZaw**

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/FaselHDProvider.kt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    private fun Element.toSearchResponse(): SearchResponse? {
+        val url = select("div.postDiv a")?.attr("href") ?: return null
+        val posterUrl = select("div.postDiv a div img")?.attr("data-src") ?:
+                        select("div.postDiv a div img")?.attr("src")
+        val title = select("div.postDiv a div img")?.attr("alt")
+            ?.replace("فيلم|مترجم|اون لاين|مسلسل|مشاهدة|انمي".toRegex(),"")
+        val quality = select(".quality").first()?.text()?.replace("1080p |-".toRegex(), "")
+        return MovieSearchResponse(
+            title!!,
```
[the remark this answers is not in the document]
Comment (LagradOst) [reply app/src/main/java/com/lagradost/cloudstream3/movieproviders/FaselHDProvider.kt:27]: or ?: throw ErrorLoadingExeption("title not found")

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/FaselHDProvider.kt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        return HomePageResponse(pages)
+    }
+
+    override suspend fun search(query: String): List<SearchResponse> {
+        val q = query.replace(" ","+")
+        val result = arrayListOf<SearchResponse>()
+        val d = app.get("$mainUrl/?s=$q").document
+            d.select("div[id=\"postList\"] div[class=\"col-xl-2 col-lg-2 col-md-3 col-sm-3\"]")
+                .mapNotNull {
+                result.add(it.toSearchResponse()!!)
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/FaselHDProvider.kt:61]: what are you doing? simply do return d.select("div[id=\"postList\"] div[class=\"col-xl-2 col-lg-2 col-md-3 col-sm-3\"]").mapNotNull { it.toSearchResponse() }

**Review by LagradOst — COMMENTED** [review]
some code quality changes

**Comment by cutterchop** [comment]
Doesn't fasel have cloud flare

**Comment by LagradOst** [comment]
bruh it has cloudflare

**Comment by ImZaw** [comment]
wait what it does? i didnt see any when i scraped it

**Comment by cutterchop** [comment]
Xd moment 2022 bruh goffy ahhhh

**Comment by odnar-dev** [comment]
faselhd use a firewall and show cloudflare only if your ip is not form an arabic country

**Comment by ImZaw** [comment]
> faselhd use a firewall and show cloudflare only if your ip is not form an arabic country

ohhh so thats why, welp that sucks i guess

**Pull request closed without merging**